src/Sylius/Bundle/CoreBundle/Validator/Constraints/ImageUniqueCodeValidator.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\Validator\Constraints;
+
+use Sylius\Component\Core\Model\ImageInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+class ImageUniqueCodeValidator extends ConstraintValidator
+{
+    /**
+     * @var RepositoryInterface
+     */
+    private $resourceRepository;
+
+    /**
+     * @param RepositoryInterface $resourceRepository
+     */
+    public function __construct(RepositoryInterface $resourceRepository)
+    {
+        $this->resourceRepository = $resourceRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($images, Constraint $constraint)
+    {
+        /** @var ImageInterface[] $imagesFromDatabase */
+       $imagesFromDatabase =  $this->resourceRepository->findAll();
+
+        /** @var ImageInterface[] $images */
+        foreach ($images as $key => $image) {
+            foreach ($imagesFromDatabase as $imageFromDatabase) {
+                if ($imageFromDatabase->getOwner() === $image->getOwner() && $imageFromDatabase->getCode() === $image->getCode()) {
+                    $this->context->addViolationAt(sprintf('[%d].code', $key), $constraint->message);
```

Comment:
`addViolationAt` is deprecated, we should use `buildViolation($message)->atPath($path)->addViolation()`.